app/controllers/media_controller.rb

```diff
@@ -0,0 +1,54 @@
+class MediaController < ApplicationController
+  before_action only: [:show, :edit, :update, :destroy, :vote] { @medium = Medium.find(params[:id]) }
+  before_action only: [ :edit, :destroy, :vote] { @type = @medium.type.pluralize }
+
+  def index
+    @media = Medium.where("type = '#{params[:type]}'")
+  end
+
+  def show; end
+
+  def new
+    @medium = Medium.new
+    @type = params[:type]
+  end
+
+  def create
+    @medium = Medium.create(media_params[:medium])
+
+    if @medium.save
+      @media_type = params[:medium][:type]
+      redirect_to polymorphic_path(@medium)
+    else
+      render "new"
+    end
+  end
+
+  def edit; end
+
+  def update
+    media_type = @medium.type.downcase.singularize
```

Comment:
The `type` is defined in a few controller methods, this could probably be assigned in a `before_action`. 

Also, just to caution you for future projects, Rails may get upset when you name a column `type` since it's considered a restricted word. Read here: http://stackoverflow.com/questions/17879024/activerecordsubclassnotfound-the-single-table-inheritance-mechanism-failed-to